static/app/components/profiling/flamegraphZoomView.tsx

```diff
@@ -782,4 +802,112 @@ const Canvas = styled('canvas')`
   position: absolute;
 `;
 
+function selectNearestFrame(n: FlamegraphFrame, dir: 'up' | 'down' | 'left' | 'right') {
```

Comment:
Do you mind adding some comments and explaining briefly what/how we expect this to behave? I can imagine this to have a lot of edge cases which makes it hard to follow.

Reply:
Will do; realized i didn't add a test for this either, which i should. it would also help serve as some self documentation

Reply:
Yeah, I would add comments somewhere that explain how we want this to behave (and maybe why). For example, I can think of a few edge cases from the top of my head that I'm not sure how they are handled (least it does not seem obvious)

- current nav is at the top of the stack and there are no children, user presses keydown, do we go to next rootmost frame?
- current nav is at the top of the stack and frame has sibling, user presses keydown, do we go to next sibling?
etc etc, I can think of a few more, so it would be nice to know the exceptions and spec this out.

Also, remember that if users switch to an inverted flamegraph, then top/down keys should be inverted

Reply:
> Also, remember that if users switch to an inverted flamegraph, then top/down keys should be inverted

💀

Reply:
fixed the inverted flamegraph behavior.

for all other cases where we've reached an end of a boundary, i try to just return the "current" highlighted node. in other words, we don't move and its a no-op.

i feel like that's just going to add a bunch of weird edge cases or bounce the user around too much

Reply:
@eliashussary Since we already allow jumping left-right we should aim on allowing users to use top/down to cycle between stacks too. I don't think implementing cycling should be too hard here (I'd be happy to pair if you want), it would be especially cool because this could allow you to use top/down to jump between stack tops and more easily identify long running stacks.